src/figuresClasses.ts

```diff
@@ -1,11 +1,93 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape;
+
+  private sides: number[];
+
+  constructor(
+    public color: Color,
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0 || a + b <= c || a + c <= b || b + c <= a) {
```

Comment:
``` a + b <= c || a + c <= b || b + c <= a ```
I threw a,b,c in array, and .sort() it, then took index 0 and 1, summed them and compared them to index 2.
Maybe my version little bit bigger, but I think, it looks better for understanding the code.🤔🤔